src/blob/index/core.rs

```diff
@@ -55,20 +55,20 @@ where
 
 #[derive(Debug, Default)] // Default can be used to initialize structure with 0
 pub(crate) struct MemoryAttrs {
```

Comment:
Please, recheck where `MemoryAttrs` is used. I see that it is only used in this module to calculate occupied memory by `InMemory` index. If it is true, then it is better to move `mem: Option<MemoryAttrs>` field from `IndexStruct` into `State::InMemory` and place it under the lock. In this way, there is not need to wrap it with `Option` + fields can be a of regular `usize` types.

Also, if my observation is correct, then `key_size`, `btree_entry_size` and `record_header_size` can be converted into `const` by making `MemoryAttrs` a generic structure, parametrized by key type

Reply:
Done, now part of `MemoryAttrs` is const, also removed unnecessary spread of functions related to this struct from other modules